### DISCUSSION TITLE
Startup crash (was: Failed to launch JVM)

### DISCUSSION THREAD

**Discussion by marpol1992** [logisim-evolution/logisim-evolution#1095] [General]
I have a problem starting the application version 3.6.0. I have jdk-16.0.2 and jre1.8.0_301 installed, but it returns an error at startup "Failed to launch JVM"

**Comment by MarcinOrlowski** [comment]
What is your exact invocation line?

**Reply by marpol1992** [reply]
I run the app from the logisim-evolution.exe file(win10)

**Reply by MarcinOrlowski** [reply]
Is there any reason you keep jre1.8.0_301 installed? That's really old version. Can you uninstall it?

**Reply by MarcinOrlowski** [reply]
Also, LSe needs Java 16. It is high chances all paths etc points to 1.8, so if you cannot uninstall, ensure JAVA_HOME or paths is pointing to v16. Alternatively you can download JAR version and launch it by hand from command line to ensure right JVM is being used. Once this works, you will where to dig further.

**Reply by marpol1992** [reply]
I uninstalled version 1.8 and launch jar version:
![image](https://user-images.githubusercontent.com/11152171/133270554-ac52e312-5c12-43a0-b70e-370f293818f4.png)

**Reply by MarcinOrlowski** [reply]
Can you please try most recent dev build and see if this problem still happens: You will find it here: https://github.com/logisim-evolution/logisim-evolution/actions/runs/1233251870

**Reply by marpol1992** [reply]
sorry but the problem is still the same in v3.7 :(

**Reply by MarcinOrlowski** [reply]
Can you please paste crash stack from dev version too (preferable, as text)?

Now, if that possible, please do NOT try to fix that at the moment because any additional action can erase the real culprit, so while I got some guesses why this happens I'd love to use your setup to confirm that instead of fixing in the blind. Will not take long. Thanks

**Reply by marpol1992** [reply]
sorry but probably can't fix it. I tried to run the application on 3 computers and it is still the same. I may be making the same mistake :P
Text crash:

```
Exception in thread "main" java.lang.ExceptionInInitializerError
        at com.cburch.logisim.prefs.AppPreferences$LocalePreference.<init>(AppPreferences.java:69)
        at com.cburch.logisim.prefs.AppPreferences.<clinit>(AppPreferences.java:454)
        at com.cburch.logisim.Main.main(Main.java:44)
Caused by: java.lang.IllegalArgumentException: Illegal pattern character 'r'
        at java.base/java.text.SimpleDateFormat.compile(SimpleDateFormat.java:846)
        at java.base/java.text.SimpleDateFormat.initialize(SimpleDateFormat.java:654)
        at java.base/java.text.SimpleDateFormat.<init>(SimpleDateFormat.java:625)
        at java.base/java.text.SimpleDateFormat.<init>(SimpleDateFormat.java:600)
        at com.cburch.logisim.util.LocaleManager.<clinit>(LocaleManager.java:212)
        ... 3 more
```

**Reply by MarcinOrlowski** [reply]
It's a bug in the app, so you do nothing incorrect. Can you please try attached build and confirm it addresses your troubles?

[logisim-evolution-3.7.0dev-all.zip](https://github.com/logisim-evolution/logisim-evolution/files/7165416/logisim-evolution-3.7.0dev-all.zip)

**Reply by marpol1992** [reply]
logisim-evolution-3.7.0dev-all.zip start correctly :)

**Comment by MarcinOrlowski** [comment]
For a record, I created a ticket #1101 related to this issue.

* #1101
